src/utils/AdaptiveTimeStep.cpp

```diff
@@ -123,7 +123,7 @@ AdaptiveTimeStep::Calculate (amrex::Real& dt, MultiBeam& beams, amrex::Real plas
         reduce_op.eval(numParticleOnTile, reduce_data,
             [=] AMREX_GPU_DEVICE (long ip) noexcept -> ReduceTuple
             {
-                if ( std::abs(wp[ip]) < std::numeric_limits<amrex::Real>::epsilon() ) return {
+                if ( wp[ip] <= 0._rt ) return {
```

Comment:
Are we just testing whether particles are discarded here? Is so, I believe we should test if `id < 0`, consistent with https://github.com/Hi-PACE/hipace/blob/fde46e2f2767530ff5598bb8477a3ef670b2f828/src/particles/pusher/GetAndSetPosition.H#L196. Not sure why we tested the weight here in the first place.

Reply:
Good point